src/commands/branches.ts

```diff
@@ -293,20 +293,48 @@ const create = async (
           if (!branch) {
             throw new Error('No default branch found');
           }
-          return { parent_id: branch.id };
+          return { parent_id: branch.id, protected: branch.protected };
         });
     }
 
     if (looksLikeLSN(props.parent)) {
-      return { parent_lsn: props.parent };
+      return props.apiClient
+        .listProjectBranches(props.projectId)
+        .then(({ data }) => {
+          const branch = data.branches.find(
+            (b) => b.parent_lsn === props.parent,
+          );
+          if (!branch) {
+            throw new Error(`Branch ${props.parent} not found`);
+          }
+          return { parent_id: branch.id, protected: branch.protected };
+        });
     }
 
     if (looksLikeTimestamp(props.parent)) {
-      return { parent_timestamp: props.parent };
+      return props.apiClient
+        .listProjectBranches(props.projectId)
+        .then(({ data }) => {
+          const branch = data.branches.find(
+            (b) => b.parent_timestamp === props.parent,
+          );
```

Comment:
Same here

Reply:
It's easier to calculate if the parent branch is protected after the new branch is created.
Changed the logic to output the warning after the branch creation and rolled back this particular change.